variant-remapping-automation/remapping_process.nf

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env nextflow
+
+def helpMessage() {
+    log.info"""
+    Remap submitted variant from one assembly version to another.
+
+    Inputs:
+            --source_assembly_accession     assembly accession of the submitted variants that needs to be remapped to the  as  accession
+            --target_assembly_accession     assembly accession of the submitted variants that needs to be remapped to the  as  accession
+            --species_name                  scientific name to be used for the species
+            --genome_assembly_dir           path to the directory where the genome should be downloaded
+            --template_properties           path to the template properties file
+    """
+}
+
+params.source_assembly_accession = null
+params.target_assembly_accession = null
+params.species_name = null
+// help
+params.help = null
+
+// Show help message
+if (params.help) exit 0, helpMessage()
+
+// Test input files
+if (!params.source_assembly_accession || !params.target_assembly_accession || !params.species_name || !params.genome_assembly_dir ) {
+    if (!params.source_assembly_accession) log.warn('Provide the source assembly using --source_assembly_accession')
+    if (!params.target_assembly_accession) log.warn('Provide the target assembly using --target_assembly_accession')
+    if (!params.species_name) log.warn('Provide a species name using --species_name')
+    if (!params.genome_assembly_dir) log.warn('Provide a path to where the assembly should be downloaded using --genome_assembly_dir')
+    exit 1, helpMessage()
+}
+
+
+
+species_name = params.species_name.toLowerCase().replace(" ", "_")
+
+
+process retrieve_source_genome {
+
+    output:
+    path "${params.source_assembly_accession}.fa" into source_fasta
+    path "${params.source_assembly_accession}_assembly_report.txt" into source_report
+
+    """
+    $params.executable.genome_downloader --assembly-accession ${params.source_assembly_accession} --species ${species_name} --output-directory ${params.genome_assembly_dir}
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.source_assembly_accession}/${params.source_assembly_accession}.fa
+    ln -s ${params.genome_assembly_dir}/${species_name}/${params.source_assembly_accession}/${params.source_assembly_accession}_assembly_report.txt
```

Comment:
I never thought of doing this to get nextflow to recognise jobs completing, I'm guessing this works better than `watchPaths`...

Reply:
I think if you know exactly the path where the file will appear it will be the best way. It's harder if you don't know what the file name will be.